sympy/matrices/expressions/matexpr.py

```diff
@@ -976,6 +989,12 @@ def transpose(self):
     def matrix_form(self):
         if self.first != 1 and self.higher != 1:
             raise ValueError("higher dimensional array cannot be represented")
+        # Remove one-dimensional identity matrices:
+        # (this is needed by `a.diff(a)` where `a` is a vector)
+        if self.first == Identity(1):
+            return self.second.T
```

Comment:
There is no test for this branch

Reply:
Line 62 of the test file.

Reply:
If you tried his diff:
```diff
$ git diff
diff --git a/sympy/matrices/expressions/matexpr.py b/sympy/matrices/expressions/matexpr.py
index 5f54bb907f..af51dd8a29 100644
--- a/sympy/matrices/expressions/matexpr.py
+++ b/sympy/matrices/expressions/matexpr.py
@@ -992,7 +992,7 @@ def matrix_form(self):
         # Remove one-dimensional identity matrices:
         # (this is needed by `a.diff(a)` where `a` is a vector)
         if self.first == Identity(1):
-            return self.second.T
+            return 1/0
         if self.second == Identity(1):
             return self.first
         if self.first != 1:
```
and all tests in sympy.matrices still passed.